### PR TITLE
[WaitForARMFeedback] Add api get_services_by_subscription of Azure Search

### PR DESCRIPTION
**Pull request by ms-xiaoy**
Add path of API "get search services by subscription" (GET/SUBSCRIPTIONS/PROVIDERS/MICROSOFT.SEARCH/SEARCHSERVICES) in the spec of API version 2015-08-19 of Azure Search (Microsoft.Search). Along with a sample.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json [hunk 1]
```diff
@@ -584,6 +584,49 @@
...
+      "get": {
+        "tags": [
+          "Services"
+        ],
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "operationId": "Services_ListBySubscription",
+        "x-ms-examples": {
+          "SearchListServicesByResourceGroup": { "$ref": "./examples/SearchListServicesBySubscription.json" }
```
Comment (abhi1509) [inline specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: is this accurate? byResourceGroup?

Comment (ms-xiaoy) [reply specification/search/resource-manager/Microsoft.Search/stable/2015-08-19/search.json]: Good catch, updated to SearchListServicesBySubscription

**Review by abhi1509 — APPROVED** [review]

**Review by bitsforthought — APPROVED** [review]

**Review by sarangan12 — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
@sarangan12 - looks like the "WaitForARMFeedback" label was missed on this one and we were not added to the reviewer list. So we missed this PR. looking now.

**Comment by ravbhatnagar** [comment]
Looks good. Signing off from ARM side.

**Comment by ms-xiaoy** [comment]
@sarangan12, please merge this PR since ARM signed off

**Comment by abhi1509** [comment]
@sarangan12 can you please merge this?

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**